### PR TITLE
op: add client endpoint format check

### PR DESCRIPTION
**Pull request by joshua-jin**

### CODE REVIEW

File: src/main/java/com/coreos/jetcd/ClientBuilder.java [hunk 1]
```diff
@@ -14,13 +14,17 @@
...
+import java.net.URL;
+import java.util.Arrays;
 import java.util.List;
 
 /**
  * ClientBuilder knows how to create an Client instance.
  */
 public class ClientBuilder implements Cloneable {
-
+  private static final List<String> SCHEMES = Arrays.asList("http", "https", "unix", "unixs");
```
Comment (lburgazzoli) [inline src/main/java/com/coreos/jetcd/ClientBuilder.java]: Is unix supported by default ?  As far as I can remember the SimpleNameResolver explicit checks for http/https. In addition the endpoint format depends on the NameResolver which is being used so maybe the check should be moved on the *NameResolver* class.

Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/ClientBuilder.java]: etcd go client supports unix and unixs mainly for testing purpose i think. Probably no need to support that for java client for now. 

cc/ @xiang90

Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/ClientBuilder.java]: @lburgazzoli for now, I think we should just accept `http/https` and get this pr merged. we can refactor the check into `NameResolver ` in a different pr after https://github.com/coreos/jetcd/issues/176 is discussed. let me know what you think.

Comment (lburgazzoli) [inline src/main/java/com/coreos/jetcd/ClientBuilder.java]: Fine with me

Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/ClientBuilder.java]: @joshua-jin please just keep schemes to `["http", "https"]`. then lgtm.

**Comment by lburgazzoli** [comment]
Can you rebase and squash ?

**Comment by joshua-jin** [comment]
@lburgazzoli may be I do some mistake in rebase process. I will close this pr and create a new one.

**Pull request closed without merging**